packages/wallets/ONTO/src/adapter.ts

```diff
@@ -0,0 +1,201 @@
+import {
+    Adapter,
+    BaseMessageSignerWalletAdapter,
+    EventEmitter,
+    scopePollingDetectionStrategy,
+    WalletAccountError,
+    WalletConnectionError,
+    WalletDisconnectedError,
+    WalletName,
+    WalletNotConnectedError,
+    WalletNotReadyError,
+    WalletPublicKeyError,
+    WalletReadyState,
+    WalletSignTransactionError,
+} from '@solana/wallet-adapter-base';
+import { PublicKey, Transaction } from '@solana/web3.js';
+
+interface ONTOWalletEvents {
+    connect(...args: unknown[]): unknown;
+    disconnect(...args: unknown[]): unknown;
+}
+
+interface ONTOWallet extends EventEmitter<ONTOWalletEvents> {
+    isONTO?: boolean;
+    publicKey?: { toBytes(): Uint8Array };
+    isConnected: boolean;
+    signTransaction(transaction: Transaction): Promise<Transaction>;
+    signAllTransactions(transactions: Transaction[]): Promise<Transaction[]>;
+    signMessage(message: Uint8Array): Promise<{ signature: Uint8Array }>;
+    connect(): Promise<void>;
+    disconnect(): Promise<void>;
+}
+
+interface ONTOWindow extends Window {
+    solana?: ONTOWallet;
```

Comment:
Please don't use `window.solana` -- we are requesting that all wallets phase it out so that they don't conflict with each other in the global namespace. `window.ontoWallet` or something would work though!

Reply:
have update